server/api.js

```diff
@@ -134,4 +135,33 @@ router.post("/publish-tenders", (req, res) => {
 	res.status(200).json({ message: "Form submitted successfully!" });
 });
 
+router.get("/tenders", async (req, res) => {
+	const { page = 1, limit = 25 } = req.query;
+	const offset = (page - 1) * limit;
+	try {
+		const result = await query(
+			`
+		SELECT
+			t.id AS "Tender ID",
+			t.title AS "Tender Title",
+			t.creation_date AS "Tender Created Date",
+			t.announcement_date AS "Tender Announcement Date",
+			t.deadline AS "Tender Closing Date",
+			t.deadline AS "Tender Project Deadline Date",
+			t.status AS "Tender Status"
+		FROM
+			tender t
+		ORDER BY
+			t.creation_date DESC
+		LIMIT $1 OFFSET $2;
+		`,
+			[limit, offset]
+		);
+
+		res.status(200).json({ tenders: result.rows });
+	} catch (err) {
+		res.status(500).json({ error: err.message });
```

Comment:
Don't send the raw error message to the client. It might contain sensitive information. Just send a generic server error code.

Reply:
I have updated it